llvm/lib/CodeGen/MachineScheduler.cpp

```diff
@@ -4263,8 +4268,9 @@ void GenericScheduler::schedNode(SUnit *SU, bool IsTopNode) {
       reschedulePhysReg(SU, true);
   } else {
     SU->BotReadyCycle = std::max(SU->BotReadyCycle, Bot.getCurrCycle());
-    BotCluster = DAG->getCluster(SU->ParentClusterIdx);
-    LLVM_DEBUG(if (BotCluster) {
+    BotClusterID = SU->ParentClusterIdx;
+    LLVM_DEBUG(if (BotClusterID != InvalidClusterId) {
```

Comment:
This would format better if you added a wrapping {} like `LLVM_DEBUG({})`

Reply:
That's a very helpful trick! Thanks!